subsys/net/ip/net_core.c

```diff
@@ -382,7 +382,7 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 		return -EINVAL;
 	}
 
-	if (!pkt->frags) {
+	if (!pkt->frags || !pkt->frags->data) {
```

Comment:
I think it's time to add a net_pkt_is_empty() function :)

Reply:
That is a good idea, I will send a new version.